### PR TITLE
Feature/parameter

### PR DESCRIPTION
**Pull request by reyoung**

### CODE REVIEW

File: python/paddle/v2/framework/graph.py [hunk 1]
```diff
@@ -125,5 +200,41 @@ def rollback(self):
...
+            'type': 'uniform_random',
+            'min': -1.0,
+            'max': 1.0
+        })
+
+        self.optimize_attr = kwargs.get('optimize_attr', {'learning_rate': 1.0})
+        self._append_initialize_ops_()
+
+    def _append_initialize_ops_(self):
+        attr = copy.deepcopy(self.init_attr)
```
Comment (Canpio) [inline python/paddle/v2/framework/graph.py]: Why do a deep copy here?

File: python/paddle/v2/framework/graph.py [hunk 2]
```diff
@@ -1,50 +1,121 @@
...
+                 lod_level=None,
+                 **kwargs):
         self.block = block
 
         if name is None:
             name = Variable._unique_var_name_()
-        self.proto = self.block.proto.new_var(name)
+        try:
+            self.desc = self.block.desc.var(name)
+            is_new_var = False
```
Comment (Canpio) [inline python/paddle/v2/framework/graph.py]: When will the constructor be called for an existing var?

File: python/paddle/v2/framework/graph.py [hunk 3]
```diff
@@ -125,5 +200,41 @@ def rollback(self):
...
+        })
+
+        self.optimize_attr = kwargs.get('optimize_attr', {'learning_rate': 1.0})
+        self._append_initialize_ops_()
+
+    def _append_initialize_ops_(self):
+        attr = copy.deepcopy(self.init_attr)
+        op_type = attr.pop('type', None)
+        block = self.block
+        assert isinstance(block, Block)
```
Comment (Canpio) [inline python/paddle/v2/framework/graph.py]: Why shall we do this `assert`?

Comment (reyoung) [reply python/paddle/v2/framework/graph.py]: Just for type hint.

File: python/paddle/v2/framework/graph.py (hunk 1 above)
Comment (reyoung) [reply python/paddle/v2/framework/graph.py]: attr and init_attr will share the same object if we do not `deepcopy` explicitly. I just keep `init_attr` unchanged.

File: python/paddle/v2/framework/graph.py (hunk 2 above)
Comment (reyoung) [reply python/paddle/v2/framework/graph.py]: Two layers share a same parameter.

**Pull request merged**